src/test/spec/retryable_reads.rs

```diff
@@ -0,0 +1,183 @@
+use bson::{doc, Bson, Document};
+
+use crate::test::{
+    assert_matches,
+    run_spec_test,
+    util::{parse_version, AnyTestOperation, EventClient, TestClient, TestEvent, TestFile},
+    CLIENT_OPTIONS,
+    LOCK,
+};
+
+const SKIPPED_OPERATIONS: &[&str] = &[
+    "count",
+    "download",
+    "download_by_name",
+    "listCollectionObjects",
+    "listDatabaseObjects",
+    "listIndexNames",
+    "listIndexes",
+    "mapReduce",
+    "watch",
+];
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn run() {
+    async fn run_test(test_file: TestFile) {
+        let has_skipped_op = test_file.tests.iter().any(|test_case| {
+            test_case
+                .operations
+                .iter()
+                .any(|op| SKIPPED_OPERATIONS.contains(&op.name.as_str()))
+        });
+        if has_skipped_op {
+            return;
+        }
+
+        for test_case in test_file.tests {
+            if let Some(skip_reason) = test_case.skip_reason {
+                println!("Skipping {}: {}", test_case.description, skip_reason);
+                continue;
+            }
+
+            let mut options = CLIENT_OPTIONS.clone();
+            if let Some(client_options) = test_case.client_options {
+                options.retry_reads = Some(client_options.get_bool("retryReads").unwrap());
```

Comment:
I'm a little concerned that we may accidentally omit other fields specified here. We may need a `pub(crate)` "merge" method for `ClientOptions` to ensure we get everything.

Reply:
The only client option that's specified for these tests is retryReads so including more options isn't totally necessary here, although I considered this when I was originally writing a more generic v2 runner. I was planning on augmenting this to support more options when I implement the v2 runner (based on this runner), but I can also incorporate it now in this commit if you think that would be better!

Reply:
Leaving for the full v2 runner sounds good to me.